lib/utils/arch.sh

```diff
@@ -21,3 +21,16 @@ ih::arch::ibrew() {
 ih::arch::mbrew() {
   /opt/homebrew/bin/brew "${@}"
 }
+
+ih::arch::get_macos_version() {
+  sw_vers -productVersion | awk -F '.' '{ printf("%d.%d\n", $1, $2) }'
+}
+
+ih::arch::is_m3_mac() {
+  local hw_model=$(sysctl -n machdep.cpu.brand_string)
+  if [[ "$hw_model" == *"M3"* ]]; then
+    return 0  # This is an M3 Mac.
+  else
+    return 1  # This is not an M3 Mac.
+  fi
```

Comment:
This could just be.

```suggestion
  [[ "$hw_model" == *"M3"* ]]
```

Reply:
Ah nice, fixed: https://github.com/ConsultingMD/homebrew-ih-public/pull/78/commits/1bf73db8f88bcbaf2121723335630e10b636f1c9